### PR TITLE
Edit/Modify data functionality

### PR DESCRIPTION
**Pull request by karakostis**
The Edit Data functionality was designed and implemented in order to give the possibility to the user to:

View data of a layer in a tabular format.
Make a simply search on the data (it uses the search functionality of Datatables.js)
Edit the data by changing its non spatial attributes.
Edit the spatial data (currently works only for points). E.g. change position of a point.
Delete a row from the layer.
Add a new feature (point, line or polygon).
Below I attach some of the use cases of the app together with screenshots:

Edit the data by changing its non spatial attributes

Click the 'Edit Attributes' button.
Edit the data by clicking on each cell.
Click the 'Save' button or click 'Cancel' to cancel the changes.

https://github.com/GeoNode/geonode/issues/4091

### CODE REVIEW

File: geonode/contrib/edit_data/templates/edit_data/wfs_add_new_line.xml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<wfs:Transaction service="WFS" version="1.1.0"
+  xmlns:wfs="http://www.opengis.net/wfs"
+  xmlns:gml="http://www.opengis.net/gml"
+  xmlns:geonode="{{ geonode_url }}">
+  <wfs:Insert>
+    <geonode:{{ layer_name }}>
```
Comment (hishamkaram) [inline geonode/contrib/edit_data/templates/edit_data/wfs_add_new_line.xml]: workspace name should be retrieved from django setting(DEFAULT_WORKSPACE )

File: geonode/contrib/edit_data/templates/edit_data/wfs_delete_row.xml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<wfs:Transaction service="WFS" version="1.0.0"
+  xmlns:ogc="http://www.opengis.net/ogc"
+  xmlns:wfs="http://www.opengis.net/wfs">
+  <wfs:Delete typeName="geonode:{{layer_name}}">
```
Comment (hishamkaram) [inline geonode/contrib/edit_data/templates/edit_data/wfs_delete_row.xml]: the same thing for the typeName

Comment (karakostis) [reply geonode/contrib/edit_data/templates/edit_data/wfs_delete_row.xml]: @hishamkaram Thanks for the comment. I have updated the code. It is not clear to me why the tests are failing. Could you please have a look as well?

**Comment by karakostis** [comment]
Thanks @afabiani 

I will look at these issues. I have one question regarding **adding specific leaflet plugins libraries in the:**

1. leaflet-plugins.min.js
2. leaflet-plugins.min.css

**I follow the below steps:**

1. add the dependencies in bower
2. run "bower cache clean"
3. run "bower update"
4. add the dependencies in grunt file
5. run "grunt production"

Although I can see the updated files with the corresponding code included, the libraries are not loaded (I am currently in the dev env).

Right now I have modified the settings.py file in order to make this work but I will also need to fix this before committing:

`if not DEBUG_STATIC:
    # if not DEBUG_STATIC, use minified css and js
    LEAFLET_CONFIG['PLUGINS'] = {
        'leaflet-plugins': {
            'js': 'lib/js/leaflet-plugins.min.js',
            'css': 'lib/css/leaflet-plugins.min.css',
            'auto-include': True,
        },
        'leaflet-areaselect': {
            'css': 'lib/css/leaflet-areaselect.css',
            'js': 'lib/js/leaflet-areaselect.js',
            'auto-include': True,
        },
    }`

**Comment by giohappy** [comment]
@karakostis AFAICS you didn't provide an alternative to the conditional check in layer_detail. I totally agree with @afabiani that contrib apps' stuff should never leak into the core code.
The same applies for the static assets. The contrib apps are in charge of serving them, they shouldn't be placed in the base folders.

PEP8 errors are still there.

**Comment by karakostis** [comment]
@giohappy thanks for the tips. I have fixed the PEP8 errors and modify the app in order to serve the static files. Regarding what @afabiani recommends, it makes sense and I will try to do this asap although right now I haven't thought of a way. Do you have any recommendations on this?

**Pull request closed without merging**